server/src/main/java/org/jboss/as/server/Main.java

```diff
@@ -448,6 +452,10 @@ private static boolean processProperties(ProductConfig productConfig, final Stri
          try {
              url = makeURL(urlSpec);
              systemProperties.load(url.openConnection().getInputStream());
+             if (systemProperties.getProperty(ServerEnvironment.HOME_DIR) != null) {
+                 // Re-create using updated properties
+                 productConfig = ProductConfig.fromFilesystemSlot(Module.getBootModuleLoader(), WildFlySecurityManager.getPropertyPrivileged(ServerEnvironment.HOME_DIR, null), systemProperties);
+             }
```

Comment:
I think the `WildFlySecurityManager.getPropertyPrivileged(ServerEnvironment.HOME_DIR, null)` can be replaced here with `systemProperties.getProperty(ServerEnvironment.HOME_DIR)` and retrieved without a privileged block.

Reply:
+1

Reply:
Fixed.